### ISSUE TITLE
After changing master - Replication map not updated

### ISSUE THREAD

**Issue by CharlotteRose** [yugabyte/yugabyte-db#7807]
**Issue**: A change in master may result in replication map errors, not reflecting new, current map for the producer. 

**Request**: Tool to easily validate/identify the producer/consumer relationship from the UI

**Comment by streddy-yb** [comment]
Hi @CharlotteRose - Can you summarize the overall scenario? Are you changing the cross-cluster replication setup across two universes?

**Comment by streddy-yb** [comment]
hi @CharlotteRose - Can you provide more context?

**Comment by CharlotteRose** [comment]
@streddy-yb - the context for this is that when using xCluster replication, if a master service is migrated off a node, that was originally in the masters list, the map does not update, and this causes numerous issues with replication and leaves the cluster in an unreliable state. 

Steps to Reproduce:
Map Values in the beginning - prior to replication 
PRIMARY 
```
[yugabyte@yb-dev-zd-1197-primary-n2 bin]$ ./yb-admin --master_addresses 10.204.0.5,10.204.0.3,10.204.0.6 get_universe_config | python -m json.tool
{
    ""clusterUuid"": ""fa98841d-6609-4e11-8d50-055befa9b535"",
    ""replicationInfo"": {
        ""affinitizedLeaders"": [
            {
                ""placementCloud"": ""gcp"",
                ""placementRegion"": ""us-east1"",
                ""placementZone"": ""us-east1-d""
            },
            {
                ""placementCloud"": ""gcp"",
                ""placementRegion"": ""us-east1"",
                ""placementZone"": ""us-east1-c""
            },
            {
                ""placementCloud"": ""gcp"",
                ""placementRegion"": ""us-east1"",
                ""placementZone"": ""us-east1-b""
            }
        ],
        ""liveReplicas"": {
            ""numReplicas"": 3,
            ""placementBlocks"": [
                {
                    ""cloudInfo"": {
                        ""placementCloud"": ""gcp"",
                        ""placementRegion"": ""us-east1"",
                        ""placementZone"": ""us-east1-d""
                    },
                    ""minNumReplicas"": 1
                },
                {
                    ""cloudInfo"": {
                        ""placementCloud"": ""gcp"",
                        ""placementRegion"": ""us-east1"",
                        ""placementZone"": ""us-east1-c""
                    },
                    ""minNumReplicas"": 1
                },
                {
                    ""cloudInfo"": {
                        ""placementCloud"": ""gcp"",
                        ""placementRegion"": ""us-east1"",
                        ""placementZone"": ""us-east1-b""
                    },
                    ""minNumReplicas"": 1
                }
            ],
            ""placementUuid"": ""ZWIzZmRlZWQtZTQ1Zi00ZWFhLThiN2MtNTU0ZGViMTAxMjUw""
        }
    },
    ""version"": 1
}
"
```
SECONDARY 
```
[yugabyte@yb-dev-zd-1197-secondary-n2 bin]$ ./yb-admin -master_addresses 10.204.0.8,10.204.0.7,10.204.0.9 get_universe_config | python -m json.tool 
{
    ""clusterUuid"": ""4c2257af-393a-4924-8c32-898a3b5608e6"",
    ""replicationInfo"": {
        ""affinitizedLeaders"": [
            {
                ""placementCloud"": ""gcp"",
                ""placementRegion"": ""us-east1"",
                ""placementZone"": ""us-east1-d""
            },
            {
                ""placementCloud"": ""gcp"",
                ""placementRegion"": ""us-east1"",
                ""placementZone"": ""us-east1-c""
            },
            {
                ""placementCloud"": ""gcp"",
                ""placementRegion"": ""us-east1"",
                ""placementZone"": ""us-east1-b""
            }
        ],
        ""liveReplicas"": {
            ""numReplicas"": 3,
            ""placementBlocks"": [
                {
                    ""cloudInfo"": {
                        ""placementCloud"": ""gcp"",
                        ""placementRegion"": ""us-east1"",
                        ""placementZone"": ""us-east1-d""
                    },
                    ""minNumReplicas"": 1
                },
                {
                    ""cloudInfo"": {
                        ""placementCloud"": ""gcp"",
                        ""placementRegion"": ""us-east1"",
                        ""placementZone"": ""us-east1-c""
                    },
                    ""minNumReplicas"": 1
                },
                {
                    ""cloudInfo"": {
                        ""placementCloud"": ""gcp"",
                        ""placementRegion"": ""us-east1"",
                        ""placementZone"": ""us-east1-b""
                    },
                    ""minNumReplicas"": 1
                }
            ],
            ""placementUuid"": ""ODljNjZjYjMtNTRjNC00ZjZjLTk0MGYtODUxMGJiOTM4Y2Jh""
        }
    },
    ""version"": 1
}"
```

1. Set up bi-directional replication 
```
[yugabyte@yb-dev-zd-1197-primary-n3 bin]$ ./yb-admin --master_addresses 10.204.0.8,10.204.0.7,10.204.0.9 get_universe_config | python -m json.tool  | grep master -C 10
{
    "clusterUuid": "4c2257af-393a-4924-8c32-898a3b5608e6",
    "consumerRegistry": {
        "producerMap": {
            "fa98841d-6609-4e11-8d50-055befa9b535": {
                "masterAddrs": [
                    {
                        "host": "10.204.0.5",
                        "port": 7100
                    },
                    {
                        "host": "10.204.0.3",
                        "port": 7100
                    },
                    {
                        "host": "10.204.0.6",
[yugabyte@yb-dev-zd-1197-primary-n3 bin]$ ./yb-admin --master_addresses 10.204.0.5,10.204.0.6,10.204.0.3 get_universe_config | python -m json.tool  | grep master -C 10
{
    "clusterUuid": "fa98841d-6609-4e11-8d50-055befa9b535",
    "consumerRegistry": {
        "producerMap": {
            "4c2257af-393a-4924-8c32-898a3b5608e6": {
                "masterAddrs": [
                    {
                        "host": "10.204.0.8",
                        "port": 7100
                    },
                    {
                        "host": "10.204.0.7",
                        "port": 7100
                    },
                    {
                        "host": "10.204.0.9",
```

2. Stop the master process on one of the nodes, that was in the masters list when replication was initiated. For this example, the master service was stopped on 10.204.0.5
```
sudo yugabyte@10.204.0.5
Last login: Tue May  4 21:15:36 2021 from 10.204.0.72
[yugabyte@yb-dev-zd-1197-primary-n3 ~]$ ps -ef | grep yb 
yugabyte  6108  6017  0 20:50 pts/1    00:00:02 python /home/yugabyte/yb-software/yugabyte-2.4.1.1-b4-centos-x86_64/bin/ycqlsh.py 10.204.0.5
yugabyte  9495     1  2 21:15 ?        00:00:04 /home/yugabyte/tserver/bin/yb-tserver --flagfile /home/yugabyte/tserver/conf/server.conf
yugabyte  9602  9495  0 21:15 ?        00:00:00 /home/yugabyte/yb-software/yugabyte-2.4.1.1-b4-centos-x86_64/postgres/bin/postgres -D /mnt/d0/pg_data -p 5433 -h 10.204.0.5 -k /tmp/.yb.9649331288882772723 -c unix_socket_permissions=0700 -c logging_collector=on -c log_directory=/mnt/d0/yb-data/tserver/logs -c shared_preload_libraries=pg_stat_statements,yb_pg_metrics,pgaudit -c yb_pg_metrics.node_name=yb-dev-zd-1197-primary-n3 -c yb_pg_metrics.port=13000 -c config_file=/mnt/d0/pg_data/ysql_pg.conf -c hba_file=/mnt/d0/pg_data/ysql_hba.conf
yugabyte 10137 10103  0 21:19 pts/0    00:00:00 grep --color=auto yb
```
3. Start the master process on a different node in that universe, for this example the master processes were started on 10.204.0.4
```
ssh yugabyte@10.204.0.4
Last login: Tue May  4 21:14:25 2021 from 10.204.0.72
[yugabyte@yb-dev-zd-1197-primary-n4 ~]$ ps -ef | grep yb
yugabyte 17162     1  0 21:14 ?        00:00:00 /home/yugabyte/master/bin/yb-master --flagfile /home/yugabyte/master/conf/server.conf
yugabyte 19837 19812  0 21:19 pts/0    00:00:00 grep --color=auto yb
yugabyte 21278     1  1 01:12 ?        00:23:04 /home/yugabyte/tserver/bin/yb-tserver --flagfile /home/yugabyte/tserver/conf/server.conf
yugabyte 21380 21278  0 01:12 ?        00:00:00 /home/yugabyte/yb-software/yugabyte-2.4.1.1-b4-centos-x86_64/postgres/bin/postgres -D /mnt/d0/pg_data -p 5433 -h 10.204.0.4 -k /tmp/.yb.17382000149456035412 -c unix_socket_permissions=0700 -c logging_collector=on -c log_directory=/mnt/d0/yb-data/tserver/logs -c shared_preload_libraries=pg_stat_statements,yb_pg_metrics,pgaudit -c yb_pg_metrics.node_name=yb-dev-zd-1197-primary-n4 -c yb_pg_metrics.port=13000 -c config_file=/mnt/d0/pg_data/ysql_pg.conf -c hba_file=/mnt/d0/pg_data/ysql_hba.conf
```

4. You will see that the map has not been updated, and does not reflect the new master 
```
[yugabyte@yb-dev-zd-1197-primary-n4 ~]$ ./master/bin/yb-admin --master_addresses 10.204.0.8,10.204.0.7,10.204.0.9 get_universe_config | python -m json.tool  | grep master -C 13
{
    "clusterUuid": "4c2257af-393a-4924-8c32-898a3b5608e6",
    "consumerRegistry": {
        "producerMap": {
            "fa98841d-6609-4e11-8d50-055befa9b535": {
                "masterAddrs": [
                    {
                        "host": "10.204.0.5",
                        "port": 7100
                    },
                    {
                        "host": "10.204.0.3",
                        "port": 7100
                    },
                    {
                        "host": "10.204.0.6",
                        "port": 7100
                    }
                ],
```
The request is to have the cluster produce an accurate map of all master addresses being used for replication.Incorrect values cause a number of issues and needs to be fixed, please  ^_^
